packages/mattermost-redux/src/reducers/entities/threads/threadsInTeam.ts

```diff
@@ -166,8 +166,10 @@ function handleLeaveChannel(state: State, action: GenericAction, extra: ExtraDat
     for (const thread of extra.threadsToDelete) {
         if (nextState[teamId]) {
             const index = nextState[teamId].indexOf(thread.id);
-            nextState[teamId] = [...nextState[teamId].slice(0, index), ...nextState[teamId].slice(index + 1)];
-            threadDeleted = true;
+            if (index !== -1) {
+                nextState[teamId] = [...nextState[teamId].slice(0, index), ...nextState[teamId].slice(index + 1)];
```

Comment:
it would be easier to just use `splice` here:

```suggestion
                nextState[teamId].splice(index, 1);
```

Reply:
Won't splice mutate the original array?

Reply:
yes, but in this case that should be fine

Reply:
I don't how it would be fine without spreading the results onto a new array 🤔

Reply:
Basically since you spread the original state onto a new array and only mutate that one it is fine, since the original one does not get mutated.

Reply:
Or if you are worried about the mutation you can also use filter

```suggestion
                nextState[teamId] = nextState[teamId].filter((_, i) => i !== index);
```

Reply:
the original state though is an object `{[string]: []string}`, so the underlying arrays will still have the same references with the original ones.
Or did you mean that since we are not mutating the original state, we won't have any issues when we get the arrays in a selector/component?

Reply:
Or even better, just use filter from the beginning.
So you can remove the if clause as well.

```suggestion
                nextState[teamId] = nextState[teamId].filter((item) => item !== thread.id);
```

as for the `threadDeleted`: are you using that to not return a new reference/signature, even if nothing changed? Does `redux` actually recognize that?

Reply:
> the original state though is an object `{[string]: []string}`, so the underlying arrays will still have the same references with the original ones.

fair enough ... then look at the solution above! :D

Reply:
> as for the threadDeleted: are you using that to not return a new reference/signature, even if nothing changed? Does redux actually recognize that?

yep, react-redux does a strict comparison, so if the reference changes then there must have been a change in state even if data hasn't changed.

Reply:
Still... the filter solution is a bit cleaner IMO, but as they both work I would not be against it to keep it that way!